### PR TITLE
Feature: add validation for contract definition id #1372

### LINKED ISSUE #1347: Add validation to avoid `ContractDefinition` id to contain `:`
# Feature Request

Creating a `ContractDefinition` with a `:` contained in the string would cause this error:
```
ContractRejectionHandler: Received contract rejection to message null. Negotiation process: 67abd883-b5b5-4e57-bf71-f5844f012a29. Rejection Reason: "Invalid id: urn:1edcc7c0-4796-46bc-bafd-87bd440cc68a:8d9e7ac4-770c-431c-9e2f-2e575c870e9a"
```

Would be good to have a validation on the `contractdefinition` api, to avoid this possibility.
Another way could be to have the "contract id parser" being able to permit the use of `:` inside the contract definition id, but seems to be more convoluted.

## Which Areas Would Be Affected?
data management api

## Why Is the Feature Desired?
To avoid exceptions

## Solution Proposal
-

## Type of Issue
improvement

## Checklist

- [x] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

### PR DESCRIPTION
**Pull request by tuncaytunc-zf**
## What this PR changes/adds

Add validation for contract definition id

## Why it does that

Validation of contract definition id by creation solves the problem in negotiation process when the id is not valid.

## Further notes

Marking jersey-bean-validation helps for the correct validation of parameters.

## Linked Issue(s)

Closes #1347 

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Comment by florianrusch-zf** [comment]
If we now would specify an incorrect ID, what would the error message look like?

**Comment by tuncaytunc-zf** [comment]
> If we now would specify an incorrect ID, what would the error message look like?

You would get "400 Bad Request"

**Comment by florianrusch-zf** [comment]
> You would get "400 Bad Request" like other validation fails.

Without any kind of error message?

**Comment by florianrusch-zf** [comment]
> You would get "400 Bad Request" like other validation fails.

Without any kind of error message?

**Comment by tuncaytunc-zf** [comment]
> > You would get "400 Bad Request" like other validation fails.
> 
> Without any kind of error message?

Unfortunately not like all other fields but as I can remember there were some discussions ongoing to improve the error handling or logging, maybe it exists already an issue for it. I think to improve the error handling was not part of this issue.

**Comment by florianrusch-zf** [comment]
@ndr-brt what do you think about this? In my opinion we should always consider to improve the error handling - also if we just fix a bug.

@tuncaytunc-zf we have since a few days the [Logging Guide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/docs/developer/logging.md). But I think that this special case is currently not be considered within the logging guide.

In addition we also discussed this kind of error handling here: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1332

File: extensions/api/api-core/build.gradle.kts [hunk 1]
```diff
@@ -28,12 +28,12 @@ dependencies {
     implementation(project(":common:util"))
     implementation("jakarta.ws.rs:jakarta.ws.rs-api:${rsApi}")
     implementation("jakarta.validation:jakarta.validation-api:${jakartaValidationApi}")
+    implementation("org.glassfish.jersey.ext:jersey-bean-validation:${jerseyVersion}") //for validation
```
Comment (ndr-brt) [inline extensions/api/api-core/build.gradle.kts:31]: This could be removed from all the extension where it's defined as `testRuntimeOnly`

**Review by ndr-brt — COMMENTED** [review]
I think this goes beyond the logging guide, as this information (failed input validation) is interesting for the client, but not for the server.
We should add some details in the response body when validation fails to make the client aware of what's needed to fix. 

Talking about the PR, this detail (the interdiction of `:` in contract definition id) should be described in the openapi specification

**Review by ndr-brt — APPROVED** [review]

File: extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/model/TransferRequestDto.java [hunk 2]
```diff
@@ -29,7 +29,6 @@ public class TransferRequestDto {
 
     @NotNull
     private String connectorAddress;
-    @NotNull
     private String id;
```
Comment (paullatzelsperger) [inline extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/model/TransferRequestDto.java:32]: Why was this removed? Transfer requests should/must always have an ID

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]

File: extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/model/TransferRequestDto.java (hunk 2 above)
Comment (ndr-brt) [reply extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/model/TransferRequestDto.java:32]: @paullatzelsperger I think this was a typo, as every system-test doesn't pass an id in the request as the EDC will generate one by itself in the transformer:

https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/60da937f68c72f471419c2c779ac1d499bc0517b/extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/transform/TransferRequestDtoToDataRequestTransformer.java#L46-L47

**Review by paullatzelsperger — APPROVED** [review]
